[BUG] Running tests fails due to missing requests dependency

**Describe the bug**
Following https://www.sktime.net/en/latest/developer_guide/continuous_integration.html#prerequisite-local-python-environment and https://www.sktime.net/en/latest/developer_guide/continuous_integration.html#full-test-suite-runs-via-pytest runs smoothly except for one test, which fails due to missing `requests` dependency:
```
========================================================================= short test summary info ==========================================================================
FAILED sktime/datasets/tests/test_datadownload.py::test_load_fpp3_public - ModuleNotFoundError: This functionality requires package 'requests' to be present in the python environment, but 'requests' was not found. To install the requirement '...
======================================= 1 failed, 4142 passed, 18142 skipped, 1 xfailed, 4 xpassed, 81 warnings in 157.64s (0:02:37) =======================================
make: *** [Makefile:31: test] Error 1
```

More details:
```
sktime/datasets/tests/test_datadownload.py:150: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
sktime/datasets/_fpp3_loaders.py:410: in load_fpp3
    y = _load_fpp3(dataset, temp_folder=None, robust=True)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        dataset    = 'olympic_running'
        temp_folder = None
sktime/datasets/_fpp3_loaders.py:360: in _load_fpp3
    _check_soft_dependencies(["requests", "rdata"])
        _check_soft_dependencies = <function _check_soft_dependencies at 0x72645b2fb880>
        dataset    = 'olympic_running'
        robust     = True
        temp_folder = None
sktime/utils/dependencies/_dependencies.py:249: in _check_soft_dependencies
    _raise_at_severity(msg, severity, caller="_check_soft_dependencies")
        _get_pkg_version_and_req = <function _check_soft_dependencies.<locals>._get_pkg_version_and_req at 0x726451016160>
        _is_str_or_tuple_of_strs = <function _check_soft_dependencies.<locals>._is_str_or_tuple_of_strs at 0x726451016d40>
        _is_version_req_satisfied = <function _check_soft_dependencies.<locals>._is_version_req_satisfied at 0x726451015f80>
        _quote     = <function _check_soft_dependencies.<locals>._quote at 0x726451016480>
        class_name = 'This functionality'
        msg        = "This functionality requires package 'requests' to be present in the python environment, but 'requests' was not found. To install the requirement 'requests', please run: `pip install requests` "
        nontrivital_bound = [False]
        normalize_reqs = True
        obj        = None
        package    = 'requests'
        package_req = ('requests',)
        package_req_strs = ["'requests'"]
        package_str = '`pip install requests`'
        package_str_q = "'requests'"
        packages   = ['requests', 'rdata']
        pkg_env_version = None
        pkg_env_versions = [None]
        pkg_names  = ['requests']
        pkg_nm     = 'requests'
        pkg_version_req = <SpecifierSet('')>
        pkg_version_reqs = [<SpecifierSet('')>]
        req_sat    = [False]
        severity   = 'error'
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

msg = "This functionality requires package 'requests' to be present in the python environment, but 'requests' was not found. To install the requirement 'requests', please run: `pip install requests` "
severity = 'error', exception_type = <class 'ModuleNotFoundError'>, warning_type = None, stacklevel = 2, caller = '_check_soft_dependencies'

    def _raise_at_severity(
        msg,
        severity,
        exception_type=None,
        warning_type=None,
        stacklevel=2,
        caller="_raise_at_severity",
    ):
        """Raise exception or warning or take no action, based on severity.
    
        Parameters
        ----------
        msg : str
            message to raise or warn
    
        severity : str, "error" (default), "warning", "none"
            whether the check should raise an error, a warning, or nothing
    
            * "error" - raises a ``ModuleNotFoundError`` if one of packages is not installed
            * "warning" - raises a warning if one of packages is not installed
              function returns False if one of packages is not installed, otherwise True
            * "none" - does not raise exception or warning
              function returns False if one of packages is not installed, otherwise True
    
        exception_type : Exception, default=ModuleNotFoundError
            exception type to raise if severity="severity"
        warning_type : warning, default=Warning
            warning type to raise if severity="warning"
        stacklevel : int, default=2
            stacklevel for warnings, if severity="warning"
        caller : str, default="_raise_at_severity"
            caller name, used in exception if severity not in ["error", "warning", "none"]
    
        Returns
        -------
        None
    
        Raises
        ------
        exception : exception_type, if severity="error"
        warning : warning+type, if severity="warning"
        ValueError : if severity not in ["error", "warning", "none"]
        """
        if exception_type is None:
            exception_type = ModuleNotFoundError
    
        if severity == "error":
>           raise exception_type(msg)
E           ModuleNotFoundError: This functionality requires package 'requests' to be present in the python environment, but 'requests' was not found. To install the requirement 'requests', please run: `pip install requests`
```

**Versions**
<details>
```
System:
    python: 3.13.5 (main, Jun 29 2025, 16:24:12) [Clang 20.1.4 ]
executable: /home/noxthot/projects/external/sktime/.venv/bin/python
   machine: Linux-6.14.0-32-generic-x86_64-with-glibc2.39

Python dependencies:
          pip: None
       sktime: 0.39.0
      sklearn: 1.7.2
       skbase: 0.12.6
        numpy: 2.2.6
        scipy: 1.15.3
       pandas: 2.3.3
   matplotlib: None
       joblib: 1.5.2
huggingface-hub: None
        numba: None
     pmdarima: None
pytorch-forecasting: None
   skforecast: None
        skpro: None
statsforecast: None
  statsmodels: None
 transformers: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None
```
</details>

--- a/sktime/datasets/tests/test_datadownload.py
+++ b/sktime/datasets/tests/test_datadownload.py
@@ -143,6 +143,10 @@ def test_load_fpp3_private():
         _ = _load_fpp3(dataset_name, temp_folder=None, robust=True)
 
 
+@pytest.mark.skipif(
+    not _check_soft_dependencies("rdata", severity="none"),
+    reason="run test only if the soft dependency rdata is installed",
+)
 @pytest.mark.datadownload
 def test_load_fpp3_public():
     """Test loading downloaded dataset from ."""

